test/test_string_map.cpp

```diff
@@ -21,6 +21,32 @@
 #include "rcutils/error_handling.h"
 #include "rcutils/types/string_map.h"
 
+class TestStringMap : public ::testing::Test
+{
+protected:
+  void SetUp() final
+  {
+    // Reset rcutil error global state in case a previously
+    // running test has failed.
+    rcutils_reset_error();
+
+    allocator = rcutils_get_default_allocator();
+    failing_allocator = get_failing_allocator();
+    string_map = rcutils_get_zero_initialized_string_map();
+  }
+
+  void TearDown() final
+  {
+    EXPECT_EQ(RCUTILS_RET_OK,
+      rcutils_string_map_fini(&string_map)) <<
+      rcutils_get_error_string().str;
```

Comment:
Doing a fini in the tear down when you don't know if an init occurred isn't a good idea imo.

Reply:
I modified this code to allow the `*_fini`to fail silently in this case as it won't leak memory.
I don't think it's worth splitting the fixture to handle those cases differently.